witwidget/notebook/colab/wit.py

```diff
@@ -270,17 +316,58 @@ def set_examples(self, examples):
     if self._rendering_complete:
       # Use BroadcastChannel to allow this call to be made in a separate colab
       # cell from the cell that displays WIT.
+      channel_start_name = 'updateExamplesStart{}'.format(self.id)
+      channel_end_name = 'updateExamplesEnd{}'.format(self.id)
       channel_name = 'updateExamples{}'.format(self.id)
+      i = 0
       output.eval_js("""(new BroadcastChannel('{channel_name}')).postMessage(
+          '')""".format(channel_name=channel_start_name))
+      while True:
+        piece = self.examples[i : i + self.SLICE_SIZE]
+        output.eval_js("""(new BroadcastChannel('{channel_name}')).postMessage(
           {examples})""".format(
-              examples=json.dumps(self.examples), channel_name=channel_name))
+              examples=json.dumps(piece), channel_name=channel_name))
+        i += self.SLICE_SIZE
+        if i > len(self.examples):
+          break
+      output.eval_js("""(new BroadcastChannel('{channel_name}')).postMessage(
+          '')""".format(channel_name=channel_end_name))
       self._generate_sprite()
 
   def infer(self):
     try:
       inferences = base.WitWidgetBase.infer_impl(self)
-      output.eval_js("""inferenceCallback({inferences})""".format(
-          inferences=json.dumps(inferences)))
+      indices = inferences['inferences']['indices'][:]
+      inferences['inferences']['indices'] = []
+      res2 = []
+      if 'classificationResult' in inferences['inferences']['results'][0]:
```

Comment:
maybe assign inferences['inferences']['results'][0] to a variable and do all the checks below on that. It looks a bit repetitive as it is.